### PR TITLE
Update pipeline to test with integrations

### PR DESCRIPTION
**Pull request by mrodm**
## What does this PR do?

Updates the pipeline with the steps needed to test the ongoing version of package-spec in a PR with all the (integration and input) packages from the [integrations repository](https://github.com/elastic/integrations)

## Why is it important?

It makes easier to test what errors if any could cause the changes in a given package-spec Pull Request

## Related issues

- Relates #530

### CODE REVIEW

**Comment by mrodm** [comment]
test integrations

**Comment by mrodm** [comment]
test integrations

File: .buildkite/hooks/pre-command [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -euo pipefail
+
+function retry {
```
Comment (jsoriano) [inline .buildkite/hooks/pre-command:5]: This function is also defined in `.buildkite/scripts/tooling.sh`, could it be imported here?

**Review by jsoriano — APPROVED** [review]

File: .buildkite/scripts/test-with-integrations.sh [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+echo "--- install jq"
+with_jq
+
+
+INTEGRATIONS_SOURCE_BRANCH=main
+INTEGRATIONS_GITHUB_OWNER=elastic
+INTEGRATIONS_GITHUB_REPO_NAME=integrations
+INTEGRATIONS_PR_BRANCH="test-${GITHUB_PR_BASE_REPO}-pr-${BUILDKITE_PULL_REQUEST}"
+INTEGRATIONS_PR_TITLE="Test ${GITHUB_PR_BASE_REPO} - DO NOT MERGE"
```
Comment (jsoriano) [inline .buildkite/scripts/test-with-integrations.sh:33]: Btw, could we also include the pull request to be tested or something in the title so not all the created PRs are titled the same?

**Review by jsoriano — APPROVED** [review]
Thanks!

**Pull request merged**